src/typescript/frontend/src/components/pages/emojicoin/components/trade-history/index.tsx

```diff
@@ -66,6 +66,18 @@ const TradeHistory = (props: TradeHistoryProps) => {
     /* eslint-disable react-hooks/exhaustive-deps */
   }, [props.data.swaps.length, swaps.length]);
 
+  const addresses = sortedSwaps.map((swap) => swap.swapper);
+  const nameResolver = useNameStore((s) => s.getResolverWithNames(addresses));
+
+  const sortedSwapsWithNames = useMemo(() => {
+    const data = sortedSwaps.map((e) => ({
+      ...e,
+      swapper: nameResolver(e.swapper),
+    }));
+    return data;
+    /* eslint-disable react-hooks/exhaustive-deps */
+  }, [nameResolver]);
+
```

Comment:
So I think the logic for this should actually just be in a hook, and then you use the hook anywhere you use a potentially resolved ANS name:

```typescript
// The hook
const useNameResolver = (address: AccountAddressInput) => {
  const normalizedAddress = AccountAddress.from(address).toString();
  const resolvedName = useNameStore((s) => s.names.get(normalizedAddress) ?? normalizedAddress);

  return resolvedName;
}

// ...then, in the body of the displaying component
const name = useNameResolver(someAddress);

// ...
<span>{name}</span>
// ...
```

And it will force a re-render only when the entry in the map changes (I'm pretty sure, at least, haha, might need to test)

This also ensures that you don't have to mutate/get rid of the swapper address in the swap data. You can keep `swapper` as is and then use the hook anywhere you need the name, usually further down the component tree

Reply:
IDK about this. The reason I did it this way is that I want to be able to pass an array to the initial function that will query the names. This is so that `localStorage` is written to only once. With your implementation, we will write to `localStorage` multiple times, and potentially slow the down the program. I think it's better to leave it this way. We can discuss this on a call.

Reply:
Discussed on eng call, agreed with you here, you can keep it as is as an array